Source/Core/InputCommon/ControllerInterface/SDL/SDL.cpp

```diff
@@ -33,43 +40,146 @@ static std::string GetJoystickName(int index)
 #endif
 }
 
-void Init()
+static void OpenAndAddDevice(int index)
 {
-#ifdef USE_SDL_HAPTIC
-  if (SDL_Init(SDL_INIT_JOYSTICK | SDL_INIT_HAPTIC) >= 0)
+  SDL_Joystick* dev = SDL_JoystickOpen(index);
+  if (dev)
   {
-    // Correctly initialized
+    auto js = std::make_shared<Joystick>(dev, index);
+    // only add if it has some inputs/outputs
+    if (!(js->Inputs().empty() && js->Outputs().size()))
```

Comment:
Should be `if (!js->Inputs().empty() && !js->Outputs().empty())`

Reply:
Woops, stupid mistake. Done.